lib/resque/scheduler.rb

```diff
@@ -177,21 +177,22 @@ def env_matches?(configured_env)
       def handle_delayed_items(at_time = nil)
         at_time ||= Time.now
         procline 'Processing Delayed Items'
-        item = nil
+        items = []
+        master = false
 
         loop do
           handle_shutdown do
-            if master?
-              item = Resque.next_delayed_item(before: at_time)
+            if (master = master?)
+              items = Resque.next_delayed_item(before: at_time, count: dequeue_batch_size)
 
-              if item
+              items.each do |item|
                 log "queuing #{item['class']} [delayed]"
                 enqueue(item)
               end
             end
           end
 
-          break if item.nil?
+          break if !master || items.empty?
```

Comment:
Why did you add this check? Wouldn't the loop break on the next iteration?

Reply:
This is a subtle bug I found while testing. 

Say the instance loses master status after assigning `items` to something non-empty (in L186), then we would never break out of the loop AFAICT.

Btw, I thought about changing L185 to be `break unless master?` but that only breaks out of the `handle_shutdown` block, not the loop